app/controllers/users/invitations_controller.rb

```diff
@@ -1,13 +1,25 @@
 class Users::InvitationsController < Devise::InvitationsController
-  before_action :authorise_manage_team, only: %i(create new)
   before_action :delete_user_record, if: :user_should_be_cleared?, only: :create
-  before_action :add_organisation_to_params, :validate_invited_user, only: :create
-
+  before_action :add_organisation_to_params, only: :create
 
 private
 
+  def authenticate_inviter!
+    redirect_to(root_path) unless current_user&.can_manage_team?
+  end
+
+  def add_organisation_to_params
+    params[:user][:organisation_id] = current_user.organisation_id
+  end
+
   def delete_user_record
-    User.find_by(email: invite_params[:email]).destroy!
+    invited_user.destroy!
+  end
+
+  def invited_user
+    @invited_user ||= User.find_by(email: invite_params[:email])
+    @invited_user = User.find_by(email: invite_params[:email]) if @invited_user&.destroyed?
+    @invited_user
```

Comment:
L21 & L22 not required.

Reply:
aye.. This was me trying to avoid some unexpected behaviour as we do have a destroy step in here.

Will remove